src/stdlib/io/file.rs

```diff
@@ -0,0 +1,323 @@
+                                        // This byte will be part of the next line
+                                    }
+                                    break;
+                                }
+                                _ => break,
+                            }
+                        }
+                    }
+                    Err(err) => return Err(err.to_string().into_value(ctx).into()),
+                }
+            }
+
+            Ok(Some(ctx.intern(&buf).into()))
+        }
+
+        "a" => {
+            let mut buf = Vec::new();
+            match file.read_to_end(&mut buf) {
+                Ok(0) => Ok(Some(ctx.intern(&[]).into())),
+                Ok(n) => {
+                    *position += n;
+                    Ok(Some(ctx.intern(&buf).into()))
+                }
+                Err(err) => Err(err.to_string().into_value(ctx).into()),
+            }
+        }
+
+        "n" => {
+            let start_position = *position;
+            let mut buf = Vec::new();
+            let mut byte = [0u8; 1];
+            let mut has_digit = false;
+
+            loop {
+                match seek_read(file, position, &mut byte) {
+                    Ok(0) => return Ok(None),
+                    Ok(_) => {
+                        if byte[0].is_ascii_whitespace() {
+                            continue;
+                        }
+                        buf.push(byte[0]);
+                        has_digit |= byte[0].is_ascii_digit();
+                        break;
+                    }
+                    Err(e) => return Err(e.to_string().into_value(ctx).into()),
+                }
+            }
+
+            loop {
+                match seek_read(file, position, &mut byte) {
+                    Ok(0) => break,
+                    Ok(_) => {
+                        if matches!(byte[0], b'0'..=b'9' | b'.' | b'-' | b'+' | b'e' | b'E') {
+                            buf.push(byte[0]);
+                            if byte[0].is_ascii_digit() {
+                                has_digit = true;
+                            }
+                        } else {
+                            break;
+                        }
+                    }
+                    Err(err) => return Err(err.to_string().into_value(ctx).into()),
+                }
+            }
+
+            if !has_digit {
+                *position = start_position;
+                return Ok(None);
+            }
+
+            let s = std::string::String::from_utf8_lossy(&buf);
+            match s
+                .parse::<i64>()
+                .ok()
+                .map(Value::from)
+                .or_else(|| s.parse::<f64>().ok().map(Value::from))
+            {
+                Some(value) => Ok(Some(value)),
+                None => {
+                    *position = start_position;
+                    Ok(None)
+                }
+            }
```

Comment:
This should use the existing number parsing logic from the lexer (like `compiler::string_utils::{read_integer, read_float}`) so that it supports Lua's number syntax.